### PR TITLE
More exit codes

### LINKED ISSUE #361: Add error codes to elemental-cli
Add errors to following parts and its dependencies:

 - [x] #362
 - [x] #363
 - [ ] #364
 - [x] #365
 - [x] #366
 - [x] #367
 - [x] #368
 - [x] #369
 - [x] #370

Example Exit codes PR: https://github.com/rancher/elemental-cli/pull/355

### LINKED ISSUE #363: Add exit codes to cloud-init

### LINKED ISSUE #364: Add exit codes to convert-disk

### LINKED ISSUE #366: Add exit codes to new

### LINKED ISSUE #367: Add exit codes to pull-image

### LINKED ISSUE #369: Add exit codes to run-stage

### PR DESCRIPTION
**Pull request by frelon**
Use elemental-error in new, run-stage, pull-image, cloud-init, convert-disk and build-iso commands.

Fixes #366, #369, #367, #363, #364, #361

Signed-off-by: Fredrik Lönnegren <fredrik.lonnegren@suse.com>

### CODE REVIEW

File: docs/elemental_exit-codes.md [hunk 1]
```diff
@@ -38,7 +38,6 @@
 | 42 | Error occurred trying to reboot|
 | 43 | Error occurred trying to shutdown|
 | 44 | Error occurred when labeling partition|
-| 44 | Error occurred when unmounting image|
```
Comment (davidcassany) [inline docs/elemental_exit-codes.md:41]: nit: no problem, as we are just coding all this right now, but we should be careful to only add errors, not modify them.

File: pkg/action/build-iso.go [hunk 2]
```diff
@@ -62,39 +63,43 @@ func NewBuildISOAction(cfg *v1.BuildConfig, spec *v1.LiveISO, opts ...BuildISOAc
...
-		return err
+		return elementalError.NewFromError(err, elementalError.CreateTempDir)
 	}
 	cleanup.Push(func() error { return b.cfg.Fs.RemoveAll(isoTmpDir) })
 
 	rootDir := filepath.Join(isoTmpDir, "rootfs")
 	err = utils.MkdirAll(b.cfg.Fs, rootDir, constants.DirPerm)
 	if err != nil {
-		return err
+		b.cfg.Logger.Errorf("Failed creating rootfs dir: %s", rootDir)
```
Comment (davidcassany) [inline pkg/action/build-iso.go:80]: thanks for the extra logging :+1:

**Review by davidcassany — APPROVED** [review]
Nice :+1:

File: docs/elemental_exit-codes.md (hunk 1 above)
Comment (frelon) [reply docs/elemental_exit-codes.md:41]: Yep, this one was a duplicate for 44 so I moved it to have it's own number instead.

**Pull request merged**